pkg/cmd/factory/http_test.go

```diff
@@ -0,0 +1,158 @@
+package factory
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewHTTPClient(t *testing.T) {
+	type args struct {
+		config     configGetter
+		appVersion string
+		setAccept  bool
+	}
+	tests := []struct {
+		name       string
+		args       args
+		envDebug   string
+		host       string
+		wantHeader map[string]string
+		wantStderr string
+	}{
+		{
+			name: "github.com with Accept header",
+			args: args{
+				config:     tinyConfig{"github.com:oauth_token": "MYTOKEN"},
+				appVersion: "v1.2.3",
+				setAccept:  true,
+			},
+			host: "github.com",
+			wantHeader: map[string]string{
+				"authorization": "token MYTOKEN",
+				"user-agent":    "GitHub CLI v1.2.3",
+				"accept":        "application/vnd.github.merge-info-preview+json",
+			},
+			wantStderr: "",
+		},
+		{
+			name: "github.com no Accept header",
+			args: args{
+				config:     tinyConfig{"example.com:oauth_token": "MYTOKEN"},
+				appVersion: "v1.2.3",
+				setAccept:  false,
+			},
+			host: "github.com",
+			wantHeader: map[string]string{
+				"authorization": "",
```

Comment:
I found this test just a tad confusing in that it made it seem like the `Accept` argument also controlled if the `authorization` header was set when it is just not set due to the config not having authorization information for `github.com`. It might be a bit less confusing if we added two tests specifically for the authorization header (one where there is config auth token value for host and one without).